src/game/KeyListener.ts

```diff
@@ -8,10 +8,11 @@ export default class KeyListener {
 
   private keyDownHandler: (key: KeyboardEvent) => void;
 
-  startListening(callback: () => any): void {
+  startListening(callback: () => void): void {
     this.keyDownHandler = (key: KeyboardEvent) => {
-      this.registerInput(key);
-      callback();
+      if (this.registerInput(key)) {
+        callback();
+      }
```

Comment:
Nice fix. I'm thinking of situations that can arise later if we have keys that are bound to actions that don't necessarily advance a turn (pause, open menu). These might be UI tasks though that are input with mouse buttons, so it might be fine. Just something to be aware of.

Reply:
Regarding the FPS counter, I ran into a similar issue in the console.log version. When the renderer is constructed, an interval is created with setInterval() which needs to be removed for hot module replacement by calling clearInterval() in stopRendering() (which is sent to module.hot.dispose()). Is it possible to dynamically delete the HTML element in a similar fashion?

Reply:
Yeah I started making the UI component so I wouldn't have to deal with that. The solution to the interval is to dispose it in the hot reload, but if I have to start doing that for every HTML element that's not scalable.